src/components/LoginForm/LoginForm.js

```diff
@@ -47,8 +47,11 @@ function findRoom(rooms, roomId) {
  * @param {Function} dispatch - the dispatch function available in the Redux store
  * @returns {Function}
  */
-function onSubmit(dispatch) {
+function onSubmit(dispatch, settings) {
   return function(data) {
+    settings.username = data.username;
```

Comment:
I would avoid mutating the `settings` object. After all, there is no reason to do that (and you are modifying an input argument). What about something like? (disclaimer: haven't tried :wink:).

```javascript
dispatch(roomActions.saveSettings({...settings, username: data.username, roomdId: data.room}).
```